Add row-wise macros (#267)
JuliaData/DataFramesMeta.jl: docs/src/index.md

* `@orderby`, for sorting data frames
* `@subset` and `@subset!`, for keeping rows of a data frame matching a given condition

> * Row-wise version of the above macros in the form of `@rtransform`, `@rtransform!`, `@rselect`, `@rselect!`, `@rorderby`, `@rsubset`, and `@rsubset!`.

Reviewer: ```suggestion
* Row-wise versions of the above macros in the form of `@rtransform`, `@rtransform!`,
  `@rselect`, `@rselect!`, `@rorderby`, `@rsubset`, and `@rsubset!`.
```

end
```

> ## Row-wise transformations with `@byrow` and `@rtransform` and friends

Reviewer: ```suggestion
## Row-wise transformations with `@byrow` and `@rtransform`/`@rselect`/etc.
```

`@byrow` provides a convenient syntax to apply operations by-row,
without having to vectorize manually. Additionally, the macros
`@rtransform`, `@rtransform!`, `@rselect`, `@rselect!`, 
`@rorderby`, `@rsubset`, and `@rsubset!` use `@byrow` by default.

DataFrames.jl provides the function wrapper `ByRow`. `ByRow(f)(x, y)`
is roughly equivalent to `f.(x, y)`. DataFramesMeta.jl allows users 
to construct expressions using `ByRow` function wrapper with the 

> syntax `@byrow` or the row-wise macros `@rtransform` etc.

Reviewer: ```suggestion
syntax `@byrow` or the row-wise macros `@rtransform`, etc.
```
